ydb/examples/topic-writer.rs

```diff
@@ -0,0 +1,78 @@
+use ydb::{
+    ClientBuilder, TopicWriter, TopicWriterMessageBuilder,
+    TopicWriterOptionsBuilder, YdbResult,
+};
+
+#[tokio::main]
+async fn main() -> YdbResult<()> {
+    let client = ClientBuilder::new_from_connection_string("grpc://localhost:2136?database=local")?
+        .client()?;
+    client.wait().await?;
+    let mut topic_client = client.topic_client();
+
+    // Default topic writer initialization
+    let writer: TopicWriter = topic_client.create_writer("/database/topic/path1".to_string());
+
+    // Parametrized topic write initialization
+    let _writer_with_params = topic_client.create_writer_with_params(
+        "/database/topic/path1".to_string(),
+        TopicWriterOptionsBuilder::default()
+            .producer_id("some_id".to_string())
+            .build()?,
+    );
+
+    // Simple write string, waits on message being written into internal buffer
+    writer
+        .write(
+            TopicWriterMessageBuilder::default()
+                .data("123".as_bytes().to_vec())
+                .build()?,
+        )
+        .await?;
+
+    // Simple write raw bytes, waits on message being written into internal buffer
+    writer
+        .write(
+            TopicWriterMessageBuilder::default()
+                .data(vec![50, 51, 52])
+                .build()?,
+        )
+        .await?;
+
+    // Write with meta info
+    writer
+        .write(
+            TopicWriterMessageBuilder::default()
+                .seq_no(123)
+                .created_at(std::time::SystemTime::now())
+                .data(vec![50, 51, 52])
+                .build()?,
+        )
+        .await?;
+
+    // Write and wait on message being sent to server and returned confirmation or error
+    let _ack_info = writer.write_with_ack(TopicWriterMessageBuilder::default()
+        .data(vec![50, 51, 52])
+        .build()?).await?;
```

Comment:
Is it possible to build return TopicWriterMessage, without error?

Reply:
I am afraid not. As far as I understand, that builder does not support compile-time validation